### ISSUE TITLE
Using stylesheets and assets from node packages

### ISSUE THREAD

**Issue by dmfrancisco** [browserify-rails/browserify-rails#161]
This is not related with Browserify but with the NPM / Sprockets integration. It's not an issue, but rather a feature request. Thank you for creating this great gem :+1: 

I wanted to require a package (`react-widgets`) that contains JavaScript, CSS and icon fonts. JavaScript worked automatically, CSS and fonts worked by simply adding `node_modules` to the assets paths. The only piece missing was solving relative URLs in the CSS files for imports and images. `bower-rails` solves this with a rake task called `bower:resolve`. I copied the code and it looks like this:

``` rake
namespace :npm do
  # Some NPM modules have relative urls in the CSS files for imports, images, etc.
  # Rails prefers using helper methods for linking to assets within CSS.
  # Relative paths can cause issues when assets are precompiled for production.
  # Based on the `rake bower:resolve` task (https://git.io/vV7KG)
  def resolve_asset_paths(root_directory)
    # Resolve relative paths in CSS
    Dir["#{root_directory}/**/*.css"].each do |filename|
      contents = File.read(filename) if FileTest.file?(filename)
      # http://www.w3.org/TR/CSS2/syndata.html#uri
      url_regex = /url\((?!\#)\s*['"]?(?![a-z]+:)([^'"\)]*)['"]?\s*\)/

      # Resolve paths in CSS file if it contains a url
      if contents =~ url_regex
        directory_path = Pathname.new(File.dirname(filename))
          .relative_path_from(Pathname.new(root_directory))

        # Replace relative paths in URLs with Rails asset_path helper
        new_contents = contents.gsub(url_regex) do |match|
          relative_path = $1
          image_path = directory_path.join(relative_path).cleanpath
          puts "#{match} => #{image_path}"

          "url(<%= asset_path '#{image_path}' %>)"
        end

        # Replace CSS with ERB CSS file with resolved asset paths
        FileUtils.rm(filename)
        File.write(filename + '.erb', new_contents)
      end
    end
  end

  desc "Resolve assets paths in node modules"
  task :resolve, :relative_directory do |_, args|
    path = args[:relative_directory] || File.join(Rails.root, "node_modules")
    resolve_asset_paths path
  end
end
```

Let me know if you are interested in supporting this in `browserify-rails`. In my opinion it's the only thing missing, because it's somewhat frequent to have `node_modules` that also contain CSS and assets.

**Comment by XXChester** [comment]
Did you fork this or how did you work around the issue and loading modules css files? I am struggling with the react-image-gallery package which has a css file.

**Comment by XXChester** [comment]
:+1:

**Comment by cymen** [comment]
I'd really recommend going to webpack directly. I contributed to browserify-rails extensively purely for my day job project which was legacy JavaScript. We used it to get from that to CommonJS and then, after I left, they finally transitioned off of the Rails asset pipeline I believe.

So I really recommend setting up webpack or browserify directly in your Procfile and abandoning the Rails asset pipeline. Or waiting yet more years for the asset pipeline to support JavaScript well. I try not to be bitter but in my opinion, the Rails asset pipeline has set back JavaScript by more than 1/2 a decade in progress in the Rails community due to not supporting basics like modules (CommonJS and beyond).

But if anyone wants to write support for it in browserify-rails and take on maintenance of it, they are more than welcome to do so. I of course have opinions however everyone is free to use this project as they wish. However, I personally am not planning on adding any more features to browserify-rails as I no longer use it in my day job and am very happy using webpack/browserify directly.
